VoodooPS2Controller/VoodooPS2Controller.cpp

```diff
@@ -158,20 +158,19 @@ void ApplePS2Controller::onWatchdogTimer()
 
 #if !HANDLE_INTERRUPT_DATA_LATER
 
-void ApplePS2Controller::handleInterrupt(PS2DeviceType deviceType)
+void ApplePS2Controller::handleInterrupt(bool watchdog)
 {
-    ////IOLog("%s:handleInterrupt(%s)\n", getName(), deviceType == kDT_Keyboard ? "kDT_Keyboard" : deviceType == kDT_Watchdog ? "kDT_Watchdog" : "kDT_Mouse");
-
     // Loop only while there is data currently on the input stream.
-    
-    bool wakeMouse = false;
-    bool wakeKeyboard = false;
+    bool wakePort[kPS2MaxIdx] = {false, false, false, false, false, false};
```

Comment:
bool wakePort[kPS2MaxIdx] {};

Otherwise it depends on kPS2MaxIdx not reducing.

Reply:
Do you mind explaining this? Not quite sure what you mean by reducing.

Reply:
Reducing the constant value. If you make it less, the code will not compile. If you increase it, it will also give a false impression of something not being initialised, which may confuse somebody. Writing the code like this gives you a named constant but forces you to remember its value implicitly, which is not right.